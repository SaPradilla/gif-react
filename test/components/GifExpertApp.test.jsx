const { render, screen, waitFor } = require("@testing-library/react")
const { GifExpertApp } = require("../../src/GifExpertApp")

describe('Pruebas en <GifExpertApp/>', () => {
  

    test('debe de hacer match con el snapshot', () => {
        const { container } = render( <GifExpertApp/> );
        expect( container ).toMatchSnapshot();

        screen.debug();
    })

    test('debe de mostrar el titulo ', () => {
        render( <GifExpertApp/> )
        expect(  screen.getByRole('heading',{level:1}) ).toBeTruthy()

    })
    
    test('debe de exitir al menos dos categoria listadas', async() => {

        render( <GifExpertApp/> )
        await waitFor(
            ()=> expect( screen.getAllByTestId('card-container').length).toBe(2)
        );

    })

})
