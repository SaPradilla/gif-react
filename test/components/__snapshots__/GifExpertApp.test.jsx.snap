// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp/> debe de hacer match con el snapshot 1`] = `
<div>
  <h1>
    GifExpert App
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar Gifs"
      type="text"
      value=""
    />
    <button
      type="submit"
    >
      Agregar
    </button>
  </form>
  <h3>
     
    Saitama
  </h3>
  <h2>
     Cargando... 
  </h2>
  <div
    class="card-grid"
    data-testid="card-container"
  />
  <h3>
     
    Goku
  </h3>
  <h2>
     Cargando... 
  </h2>
  <div
    class="card-grid"
    data-testid="card-container"
  />
</div>
`;
